lua/xcodebuild/project/manager.lua

```diff
@@ -41,6 +41,58 @@ local function validate_xcodeproj_tool()
   return true
 end
 
+---Searches for a `.xcodeproj` file in the given directory.
+---
+---Uses the `find` command to search for the first `.xcodeproj` file
+---in the specified directory and returns its absolute path if found.
+---
+---@param dir string The directory to search in.
+---@return string|nil The absolute path to the `.xcodeproj` file, or nil if not found.
+local function find_xcodeproj_file(dir)
+  local function shell_escape(str)
+    return "'" .. string.gsub(str, "'", "'\\''") .. "'"
+  end
+
+  local cmd = "find " .. shell_escape(dir) .. ' -maxdepth 1 -name "*.xcodeproj" -print -quit 2> /dev/null'
```

Comment:
I believe -quit makes it quit the search after the first match, and if -quit is used, then the result is not printed to stdout unless you add -print

Reply:
Interesting 👍 I tested it and seems you are right, but I also discovered a very weird behavior. `-print` must be before `-quit`.

Also, there is one problem here, if find doesn't have access to some folder it will fail. However, I don't think we need to be worried about that inside the project repo.

Reply:
I think it just does them left to right, for example if you run `find ./ -maxdepth 1 -name "*.xcodeproj" -print -print -print -quit`, it will print first match three times and then quit. So when you put -quit -print it just quits and never gets to printing